runtime/dart_controller.cc

```diff
@@ -105,6 +105,13 @@ bool DartController::SendStartMessage(Dart_Handle root_library) {
   return LogIfError(result);
 }
 
+void DartController::RunFromKernel(const uint8_t* buffer, size_t size) {
+  tonic::DartState::Scope scope(dart_state());
+  LogIfError(Dart_LoadKernel(Dart_ReadKernelBinary(buffer, size)));
+  if (SendStartMessage(Dart_RootLibrary()))
+    exit(1);
```

Comment:
Sounds reasonable. 

Can someone point me at code that is properly exiting on error? Is `RunFromSource` a good example? If yes, then that means all of the call sites need to be updated as well to handle an error return value.

This sounds like a more involved change, so I'm happy to clean this up in a subsequent PR along with`RunFromPrecompiledSnapshot` and`RunFromSnapshot` (which were already calling`exit`).